javalib/src/main/scala/java/io/OutputStreamWriter.scala

```diff
@@ -45,7 +45,11 @@ class OutputStreamWriter(private[this] var out: OutputStream,
     this(out, Charset.defaultCharset)
 
   def this(out: OutputStream, charsetName: String) =
-    this(out, Charset.forName(charsetName))
+    this(out, try {
```

Comment:
consider putting the try as a separate statement above:

```scala
val charset = try {
 ...
} catch {
  ...
}
```

nested like this is a little funky IMHO (@sjrd has the last word of course).

Reply:
Unfortunately in this case there's no choice, because Scala won't accept anything before the `this()` constructor call. So the way @exoego did it here is correct. (However, yes, in general, it would be better with a `val`.)

Reply:
As explained by @sjrd, this is a trick to convince Scala compiler.